### DISCUSSION TITLE
Convert from sRGB to another ICC Profile

### DISCUSSION THREAD

**Discussion by Tykester66** [dlemstra/Magick.NET#762] [Help]
I'm trying to convert from the SRGB color profile to another profile. I've tried to code below. It is close to what I want, but doesn't quite give me the output I want.

MagickImage image = new MagickImage(@"old.bmp");
image.TransformColorSpace(ColorProfile.ColorMatchRGB, new ColorProfile(@"C:\Windows\System32\spool\drivers\color\new.icc"));
image.Strip();
image.Write(@"new.bmp");

First, am I doing this the right way?

Second, when I state it is close, I'm comparing this against another application's conversion to the new.icc. Without the conversion, most OLD pixels do not match the 3rd party conversion's output (~61% are not equal). After applying this code above, the NEW pixels are close. There are only ~1% that are different.    Is this because the underlying transformation routines are (likely) different between the 3rd party app and ImageMagick?

**Comment by dlemstra** [comment]
This is indeed the way change the colors from one profile to another profile. And yes it is very likely that the difference is due to the underlying implementation. Another reason could be the pixel depth. This depends on the version of Magick.NET they could be using Q8 while you are using Q16.

p.s. The `MagickImage` class is `IDisposable` make sure you add a `using`.
